packages/lesswrong/server/repos/PageCacheRepo.ts

```diff
@@ -0,0 +1,38 @@
+import type { CompleteTestGroupAllocation } from "../../lib/abTestImpl";
+import PageCache from "../../lib/collections/pagecache/collection";
+import { getClientBundle } from "../utils/bundleUtils";
+import AbstractRepo from "./AbstractRepo";
+
+export type MeanPostKarma = {
+  _id: number,
+  meanKarma: number,
+}
+
+export default class PageCacheRepo extends AbstractRepo<DbPageCacheEntry> {
+  constructor() {
+    super(PageCache);
+  }
+
+  async lookupCacheEntry({path, completeAbTestGroups}: {path: string, completeAbTestGroups: CompleteTestGroupAllocation}): Promise<DbPageCacheEntry | null> {
```

Comment:
"complete" A/B test groups is the full set of groups assigned to the user. When we store the cache entry (locally or in the db), we only store the "relevant" A/B test groups, these are the ones which were actually used in the render path via `useABTest`. Doing it this way makes us way more likely to get a cache hit, because most render paths don't actually use most of the A/B tests that we have set up